handlers/deposit_resource.go

```diff
@@ -47,33 +48,16 @@ func (d *depositResourceEntry) Handle(params operations.DepositResourceParams) m
 	return operations.NewDepositResourceCreated().WithPayload(response)
 }
 
-func (d *depositResourceEntry) persistResource(resourceID string, params operations.DepositResourceParams) error {
-	resource := d.persistableResourceFromParams(resourceID, params)
-	return d.rt.Repository().CreateItem(resource)
-}
-
-func (d *depositResourceEntry) persistableResourceFromParams(resourceID string, params operations.DepositResourceParams) *persistence.Resource {
-	resource := &persistence.Resource{ID: resourceID}
-	resource.Access = *params.Payload.Access
-	resource.AtContext = *params.Payload.AtContext
-	resource.AtType = *params.Payload.AtType
-	resource.Label = *params.Payload.Label
-	resource.Preserve = *params.Payload.Preserve
-	resource.Publish = *params.Payload.Publish
-	resource.SourceID = params.Payload.SourceID
-	return resource
-}
-
-func (d *depositResourceEntry) addToStream(id *string) error {
-	message, err := json.Marshal(id)
-	if err != nil {
-		return err
-	}
-	if d.rt.Stream() == nil {
-		log.Printf("Stream is nil")
-	}
-	if err := d.rt.Stream().SendMessage(string(message)); err != nil {
-		return err
+func (d *depositResource) loadParams(id string, params operations.DepositResourceParams) interface{} {
+	// NOTE: This section will be replaced by DataUtils
```

Comment:
dataUtils or just passing the full `map[string]interface{}` instead (which seems to work)

Reply:
Agreed.